types/create-test-server/create-test-server-tests.ts

```diff
@@ -19,7 +19,7 @@ const server = createTestServer().then((server) => {
     });
 
     // You can return a body directly too
-    server.get("/foo", () => "bar");
+    server.get("/foo", () => void "bar");
```

Comment:
This kind of change seems odd; the comment above implies that you can return a body, but this test is making it return `undefined`.

Reply:
yeah, it's because you actually can not, handlers should return nothing — `void` — not return at all.
But technically, `void` excuses returning anything, but not `void | Promise<void>`.

Reply:
But, https://www.npmjs.com/package/create-test-server documents this behavior, and explicitly contains code to make this work: https://github.com/lukechilds/create-test-server/blob/2357c501af576e0f12226f105b39de5f61b2c21a/src/index.js#L30

I don't think this change is accurate.

Reply:
If anything, this may mean the package was wrongly typed and shouldn't depend on express in full?

Reply:
No, I'm not a maintainer of this package, I'm on the TS team and a DT maintainer, reviewing things because this PR affects at least one critical package and also affects more than one package.

Reply:
What would you recommend, @jakebailey ?

Reply:
> If anything, this may mean the package was wrongly typed and shouldn't depend on express in full?

Probably, but at the moment, such wrongly typed package blocks making types for the new version of Express.
Your expertise as TS Team member and DT maintainer could be very helpful to solve this puzzle, @jakebailey

Reply:
I've got a better solution...

Reply:
I would expect the fix would involve fixing the types for the package itself so that changes to express do not affect it, but I have not yet had time to look at this.

Given this PR is touching a load of packages, some of which _also_ look like they involve return values, I would really want to read each of the packages to understand if the changes are correct. The same error may have been made there.

Reply:
Thank you, @jakebailey .
I reverted that change of the test and replaced it with a correction of the package type involved in the problematic test: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70563/files#r1771956407

Reply:
> I would expect the fix would involve fixing the types for the package itself

I'd love to limit my PR to `express` only if you could tell me, as TS development team member, an alternative way to say `void | Promise<void>` while preserving the same compatibility to `any` as pure `void` provides according to this article, @jakebailey :

https://www.typescriptlang.org/docs/handbook/type-compatibility.html#any-unknown-object-void-undefined-null-and-never-assignability

Reply:
One way I've seen this solved is to write `(() => void) | (() => Promise<void>)`, but that would have to be a change in `express`.

https://github.com/jakebailey/hereby/blob/main/src/index.ts#L27

[Playground Link](https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwAspVgIQAKIksmALnnIEp4BeAPngDccth4AfeAAUYOALZYAziAA83Xm0b15wANwBYAFBatVUhSasOAby3xz8OBmQx8ARgBMAZgAsGzQF9GOzaEiwEFHRsPEJifQdKcJp6ckN2Lh5gZkE45gSRcSlZFUUlRN53Hz0ySPiTMwsrG3tnNy0vIA)

But I have definitely seen `void | Promise<void>` more often... react has that type in a few places.

I can't say that "solving" that is a big deal, though; isn't returning something pretty much always an error, because it implies `express` will actually do something with the result?